pkg/controller/statefulset/stateful_set.go

```diff
@@ -130,6 +136,26 @@ func NewStatefulSetController(
 	ssc.podLister = podInformer.Lister()
 	ssc.podListerSynced = podInformer.Informer().HasSynced
 
+	if err := podInformer.Informer().AddIndexers(cache.Indexers{
+		StatefulSetIndex: func(obj interface{}) ([]string, error) {
+			pod, ok := obj.(*v1.Pod)
+			if !ok {
+				return nil, nil
+			}
+			if ref := metav1.GetControllerOf(pod); ref != nil {
+				if ref.Kind == controllerKind.Kind {
+					return []string{string(ref.UID)}, nil
+				}
+				// Ignore pods controlled by other controllers
+				return nil, nil
+			}
+			return []string{OrphanPodIndexKey}, nil
```

Comment:
we always return the same string? `"Orphan"`

Reply:
comparing with the following, where they are finding something specific to return
- https://github.com/kubernetes/kubernetes/blob/129661b2c7a31edbf48a38c11cff6e5eeb275411/pkg/controller/replicaset/replica_set.go#L177
- https://github.com/kubernetes/kubernetes/blob/129661b2c7a31edbf48a38c11cff6e5eeb275411/pkg/controller/nodelifecycle/node_lifecycle_controller.go#L400

Reply:
can you please add some notes on why we are doing this?

Reply:
The idea is to be able to index orphan pod even though they don't have an ownerRef UID.
If we don't return a key, indexing won't happen for orphaned pods.

Reply:
got it!

Reply:
Yes, @mengqiy understood it correctly, that is the intent of `Orphan Index` . These will be later used to see if given Statefulset can Claim/Adopt it as its parent in reconciler loop.